source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -8,55 +8,20 @@
 #include <MaterialXGenShader/GenContext.h>
 #include <MaterialXGenOgsXml/OgsXmlGenerator.h>
 
-MaterialXData::MaterialXData(   mx::DocumentPtr document,
-                                const std::string& elementPath,
-                                const MaterialX::FileSearchPath& librarySearchPath )
-    : _document(document)
+MaterialXData::MaterialXData(mx::DocumentPtr document,
+                             mx::ElementPtr element,
+                             const MaterialX::FileSearchPath& librarySearchPath ) :
+    _document(document),
+    _element(element)
 {
     if (!_document)
     {
         throw mx::Exception("No document specified");
     }
 
-    std::vector<mx::TypedElementPtr> renderableElements;
-    mx::findRenderableElements(_document, renderableElements);
-
-    // Nothing specified. Find the first renderable element and use that
-    if (elementPath.empty())
-    {
-        if (renderableElements.empty())
-        {
-            throw mx::Exception(
-                "No element path specified and no renderable elements in the document."
-            );
-        }
-
-        _element = renderableElements.front();
-    }
-    else
+    if (!_element)
```

Comment:
Remove "hidden" determination logic from here. Simply if you give a null element this fails.